---
title: "Cross-Ancestry TWAS Project Writeup"
author: "Riley Xin"
date: "2024-01-15"
output: html_document
---
## Overview

Transcriptome-wide association studies (TWAS), combined with gene-based fine mapping, enable the identification of potential causal genes for diseases. TWAS requires an expression prediction model (or eQTL data) and a GWAS dataset with matched ancestry. In practice, performing TWAS for underrepresented populations is challenging because GWAS data with matching ancestry are often unavailable. Even when available, non-European (non-EUR) GWAS suffer from small sample sizes, limiting the power for TWAS to detect potential causal genes. 

To identify candidate causal genes for asthma, we perform cTWAS analyses using different combinations of GWAS and eQTL datasets. The goal is to prioritize genes with high posterior inclusion probabilities (PIP) and conduct downstream functional analyses, independent of ancestry-related concerns or model optimization at this stage. After establishing these findings, we can explore strategies to build our own gene expression prediction model that will enable TWAS in ancestry-mismatched settings.

## cTWAS Analysis on Asthma

We performed cTWAS using various sets of GWAS datasets (unpublished COA, AOA and [GTEx](https://zenodo.org/records/3629742#.XjCh9OF7m90)) and eQTL ([eQTLGen](https://www.nature.com/articles/s41588-021-00913-z), URECA, [1K1K](https://www.science.org/doi/10.1126/science.abf3041), [DICE](https://www.science.org/doi/10.1126/sciimmunol.abm2508)). A summary table of the datasets and cTWAS results is provided below. The new eQTL dataset, URECA, is derived from an from African American population, while all other GWAS/eQTL datasets are from the Europearn population. Asthma eQTLs are known to explain only a small portion of heritability, which may limit our ability to detect strong signals. We are interested in genes with high PIP for functional analysis. 

```{r echo=FALSE, message=FALSE, warning=FALSE}
library(kableExtra)

data_table <- data.frame(
  "Analysis" = c("1","2","3", "4"),
  "GWAS" = c("AOA", "COA", "COA", "AOA"),
  "GWAS n(c/ctrl.)" = c("20,702/308,537", "9,613/308,537", "9,613/308,537", "20,702/308,537"),
  "Gwas population" = c("White British", "White British", "White British", "White British"),
  "eQTL" = c("eQTLGen", "eQTLGen", "URECA", "URECA"),
  "eQTL n" = c("31684", "31684", "300", "300"),
  "eQTL tissue" = c("Blood", "Blood", "Nasal Epithelial", "Nasal Epithelial"),
  "eQTL population" = c("EUR", "EUR", "AFR", "AFR"),
  "LD population" = c("EUR", "EUR", "EUR", "EUR")
)

knitr::kable(data_table, format = "html", escape = FALSE) %>%
  kable_styling(full_width = FALSE, bootstrap_options = c("striped", "hover")) %>%
  column_spec(1:ncol(data_table), width = "auto")

ctwas_table <- data.frame(
  "Analysis" = c("1","2","3", "4"),
  "prior group shared" = c("True", "True", "True", "True"),
  "top QTL" = c("True", "True", "True", "True"),
  "total n genes" = c("16987", "16987", "3638", "3638"),
  "n genes after harmonization" = c("16703", "16703", "3081", "3081"),
  "genes PIP greater 0.8" = c("ENSG00000197536", "ENSG00000106799, ENSG00000138160, ENSG00000134954, ENSG00000265148", "NA", "NA")
)

knitr::kable(ctwas_table, format = "html", escape = FALSE) %>%
  kable_styling(full_width = FALSE, bootstrap_options = c("striped", "hover")) %>%
  column_spec(1:ncol(ctwas_table), width = "auto")
```

(I'll plot results in a single panel once we run all datasets)
```{r echo=FALSE}
knitr::include_graphics("assets/ctwas_convergence_plots.png", error = FALSE)
```


## Existing Approaches

Current research addresses ancestry-related issues in TWAS by adjusting GWAS summary statistics or eQTL weights, focusing on scenarios with either multiple GWAS and eQTL from different ancestries are available, or multiple GWAS and a single-ancestry eQTL data is available. Below we review relevant approaches:  

### SBayesRC

Prior on GWAS weights based on functional annotations that is adaptable to population mismatches; does not require training on data from the target population to make predictions...

### TESLA

[TESLA](https://www.nature.com/articles/s41588-022-01282-x "Multi-ancestry transcriptome-wide association analyses yield insights into tobacco use biology and drug repurposing") works for the case with multiple GWAS and a single eQTL dataset. The idea is to combine the GWAS weights in a way that leverages ancestry information based on allele frequency principal components (PCs). 

TWAS statistics are normally formulated as:
\[
U_{\text{TWAS}} = \sum_j w_j z_j,
\]
where \( w_j \) are the eQTL-derived weights and \( z_j = b_j / s_j \) are the GWAS \( z \)-scores computed from phenotypic effect estimates \( b_j \) and their standard errors \( s_j \). TWAS tests yield optimal power when the eQTL weights are proportional to the phenotypic weights, which is often violated when the data have mismathced ancestry.

The method tries to adjust \( b_j \) to be the predicted phenotypic effect \( \hat{b}_j^{[L]} \), modeled as a linear combination of the phenotypic effects across all participating studies.

**First**, construct the allele frequency (AF) matrix \( F \), where each row represents a study and each column represents a shared genetic variant. The AF matrix \( F \) is derived from allele frequency data assumed to be available in the GWAS summary statistics. Perform SVD on \( F = U S V^\top.\)
To project \( F \) onto the first \( L \) PCs, calculate: \( F V^{[L]} = [US]_{1:L},\) where \( V^{[L]} \) represents the first \( L \) PCs, and denote this as \( X_l \) for \( l = 0, \ldots, L \).

The phenotypic effects of genetic variants in study \( k \) are modeled as a linear combination of the effects from each PC:
\[
b_{jk} = \sum_{l=0}^{L} X_{lk} \gamma_{lj} + \epsilon_{jk},
\]
where \( b_{jk} \) follows a normal distribution with variance \( s^2_{jk} \).

The regression coefficients for variant \( j \) across all PCs are estimated using weighted least squares:
\[
\hat{\gamma}_j^{[L]} = \left(X^{[L]\top} \Omega_j X^{[L]}\right)^{-1} X^{[L]\top} \Omega_j b_j,
\]
where \( \Omega_j = \text{diag}(s_{j1}, \ldots, s_{jK}) \). The meta-regression coefficients \( \hat{\gamma}_j^{[L]} \) quantify the extent of SNP effect heterogeneity as a function of ancestry. 

**Next**, using \( \hat{\gamma}_j^{[L]} \), the phenotypic effects in the ancestry of the eQTL dataset can be estimated by:
\[
\hat{b}_j = \bar{X}^{[L]} \hat{\gamma}_j^{[L]},
\]
where \( \bar{X}^{[L]} \) is the projected ancestry coordinates of the eQTL dataset in the PC space defined by the GWAS studies:
\[
\bar{X}^{[L]} = \bar{F} V^{[L]}.
\]
Now, the proportionaly condition is satisfied. The covariance matrix of \( \hat{b}_j \) can be estimated using LD matrix from the ancestry of each study.

### METRO

[METRO](https://pmc.ncbi.nlm.nih.gov/articles/PMC9118130/ "METRO: Multi-ancestry transcriptome-wide association studies for powerful gene-trait association detection") requires multiple GWAS and eQTL datasets from different ancestries, either at the summary statistics level or the individual-level. It jointly models SNP effects on expression and phenotype by: 1. Assuming that SNP effects on gene expression in the GWAS can be expressed as a weighted summation of the SNP effects on gene expression in expression studies performed across different ancestries, and regressing phenotype on genotype $\times$ SNP effects. 2. A non-Bayesian likelihood-based approach with EM for parameter estimation.

## Methods

Existing approaches all assume multiple GWAS input and pull information across datasets to account for ancestral differences. However, we are interested in the case of a single GWAS dataset and an eQTL dataset from different ancestries and finding causal genes in the ancestry of the eQTL dataset. Since GWAS effect sizes are directly influenced by allele frequencies, a potential solution is to adjust the GWAS weights  by allele frequency to be on the same scale with the eQTL weights. This involves modeling GWAS weights as a function of allele frequency...





